mafContrib/TDEsPopMetric.py

```diff
@@ -0,0 +1,125 @@
+
+        # Let's organize the data in to a list of dicts for easy lookup
+        self.data = []
+        filternames = 'ugrizy'
+        for lc in lcs:
+            new_dict = {}
+            for filtername in filternames:
+                infilt = np.where(lc['filter'] == filtername)
+                new_dict[filtername] = lc[infilt]
+            self.data.append(new_dict)
+
+    def interp(self, t, filtername, lc_indx=0):
+        """
+        t : array of floats
+            The times to interpolate the light curve to.
+        filtername : str
+            The filter. one of ugrizy
+        lc_index : int (0)
+            Which file to use.
+        """
+
+        result = np.interp(t, self.data[lc_indx][filtername]['ph'],
+                           self.data[lc_indx][filtername]['mag'],
+                           left=99, right=99)
+        return result
+
+
+class TDEsPopMetric(metrics.BaseMetric):
+    def __init__(self, metricName='TDEsPopMetric', mjdCol='observationStartMJD', m5Col='fiveSigmaDepth',
+                 filterCol='filter', nightCol='night', ptsNeeded=2, file_list=None, mjd0=59853.5,
+                 **kwargs):
+        maps = ['DustMap']
+        self.mjdCol = mjdCol
+        self.m5Col = m5Col
+        self.filterCol = filterCol
+        self.nightCol = nightCol
+        self.ptsNeeded = ptsNeeded
+
+        self.lightcurves = tde_lc(file_list=file_list)
+        self.mjd0 = mjd0
+
+        waveMins = {'u': 330., 'g': 403., 'r': 552., 'i': 691., 'z': 818., 'y': 950.}
+        waveMaxes = {'u': 403., 'g': 552., 'r': 691., 'i': 818., 'z': 922., 'y': 1070.}
+
+        self.a = {}
+        self.b = {}
+        for filtername in waveMins.keys():
+            testsed = Sed()
+            testsed.setFlatSED(wavelen_min=waveMins[filtername],
+                               wavelen_max=waveMaxes[filtername],
+                               wavelen_step=1)
+            self.a[filtername], self.b[filtername] = testsed.setupCCM_ab()
+        self.R_v = 3.1
+
+        cols = [self.mjdCol, self.m5Col, self.filterCol, self.nightCol]
+        super(TDEsPopMetric, self).__init__(col=cols, units='Detected, 0 or 1',
+                                            metricName=metricName, maps=maps,
+                                            **kwargs)
+
+    def run(self, dataSlice, slicePoint=None):
+        result = 0
+        t = dataSlice[self.mjdCol] - self.mjd0 - slicePoint['peak_time']
+        mags = np.zeros(t.size, dtype=float)
+
+        for filtername in np.unique(dataSlice[self.filterCol]):
+            infilt = np.where(dataSlice[self.filterCol] == filtername)
+            mags[infilt] = self.lightcurves.interp(t[infilt], filtername, lc_indx=slicePoint['file_indx'])
+            # Apply dust extinction on the light curve
+            A_x = (self.a[filtername][0]+self.b[filtername][0]/self.R_v)*(self.R_v*slicePoint['ebv'])
+            mags[infilt] -= A_x
```

Comment:
Why aren't you using the methods on Sed to calculate the magnitude and extinction, but instead calculating it separately here? With this shortcut here, you're not including the SED itself (which seems reasonable because you probably don't have that information, given the lightcurve is just a magnitude) -- so you could take the first part of this A_x calculation ((self.a[filtername][0]+self.b[filtername][0]/self.R_v) and just do it once per filter in the init as well (maybe adding a bit of comment like so ...
```
     # Calculate the inputs for dust extinction (in magnitudes) across each filter
     # Dust extinction at a particular wavelength is calculated as : 
     #  A_lambda = (a_x * R_v + b_x ) * E(B-V)   
     #    ... E(B-V) comes from our extinction maps & R_v is taken as a canonical 3.1
     # A_lambda, integrated over the bandpass, reflects the magnitudes of extinction to apply. 
     # We only have lightcurves in magnitudes (not full SEDs), so just calculate A_filter/E(B-V) (==AE) here  
     # (since we don't know E(B-V) yet and will have to adjust below)
        self.AE = {}
        self.R_v = 3.1
        for filtername in waveMins:   
            testsed = Sed()
            testsed.setFlatSED(wavelen_min=waveMins[filtername],
                               wavelen_max=waveMaxes[filtername],
                               wavelen_step=1)
            a, b = testsed.setupCCM_ab()
            self.AE[filtername] = testsed.addDust(a_x=a, b_x=b, ebv=1, R_v=self.R_v)
         #  (then dust extinction for our objects below is self.AE[filtername] * actual E(B-V)   
```

(ok, the math SEEMS right to me but  could definitely stand for some double checking I didn't skip a log or something) 
.. also, if we're not giving people the option to change R_v, then might not be worthwhile to make it a proper attribute since it's no longer reused later.

Reply:
yeah, this is copy/pasta from the exgalM5 metric. We should pull in all into a utility function that's smarter and updates with the throughputs.